kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/FileSystemDataset.java

```diff
@@ -562,4 +572,35 @@ public boolean isEmpty() {
     }
   }
 
+  @Override
+  public void signalReady() {
+    unbounded.signalReady();
+  }
+
+  @Override
+  public boolean isReady() {
+    return unbounded.isReady();
+  }
+
+  public void signalReady(Constraints viewConstraints) {
+    Preconditions.checkNotNull(viewConstraints, "Constraints cannot be null");
+
+    Path signalsPath = new Path(directory, SIGNALS_DIRECTORY_NAME);
+
+    SignalManager signalManager = SignalManager.create(fileSystem.getConf(), signalsPath);
```

Comment:
It's a little awkward to have dataset call `signalReady` on unbounded, and then have unbounded call `signalReady` on the dataset. I think a cleaner way to do this is to have a package-local `getSignalManager` that everything can use via `dataset.getSignalManager()`. Then the dataset can lazily create the signal manager and also cache the instance.